### PR TITLE
Fixes for all the test bugs

### PR DESCRIPTION
**Pull request by shruacha1234**
Signed-off-by: Shruthi <Shruthi.Acharya@ibm.com>

### CODE REVIEW

File: tests/scripts/common_functions.sh [hunk 1]
```diff
@@ -174,8 +177,8 @@ function deploy_autotune() {
 	${cmd}
 	
 	status="$?"
-	# Check if autotune is deployed 
-	if [ "${status}" -eq "1" ]; then
+	# Check if autotune is deployed(Ignore in case of configmap_yaml_tests)
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:180]: Include a space here before braces and elaborate why it needs to be ignored for config map yaml tests

File: tests/scripts/common_functions.sh [hunk 2]
```diff
@@ -109,6 +111,7 @@ function update_yaml() {
 # Set up the autotune
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:111]: Include input parameters

File: tests/scripts/configmap_yaml_tests.sh [hunk 3]
```diff
@@ -88,7 +89,7 @@ function configmap_yaml_tests() {
 			
 			#create autotune setup
 			echo -n "Deploying autotune..."| tee -a ${LOG}
-			setup ${LOG_DIR} >> ${SETUP_LOG} 2>&1
+			setup ${LOG_DIR} "1" >> ${AUTOTUNE_SETUP_LOG} 2>&1
```
Comment (chandrams) [inline tests/scripts/configmap_yaml_tests.sh:92]: Use a variable instead of directly specifying "1"

File: tests/scripts/common_functions.sh [hunk 4]
```diff
@@ -174,10 +174,10 @@ function deploy_autotune() {
 	${cmd}
 	
 	status="$?"
-	# Check if autotune is deployed 
-	if [ "${status}" -eq "1" ]; then
+	# Check if autotune is deployed. Ignore the status check if ignore_deployment_status_check is set to "1". In case of configmap yaml tests we need not check if autotune has deployed properly during the setup since it is done as part of the test.
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:177]: Comment is too long split into multiple lines

File: tests/scripts/common_functions.sh [hunk 5]
```diff
@@ -555,7 +557,7 @@ function run_test() {
 	echo ""
 	
 	# Delete the prometheus service
-	kubectl delete svc prometheus-svc -n monitoring
+	kubectl delete svc prometheus-test -n monitoring
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:560]: Replace monitoring with NAMESPACE

File: tests/scripts/common_functions.sh [hunk 6]
```diff
@@ -1444,14 +1446,14 @@ function get_autotune_pod_log() {
 # Expose prometheus as nodeport and get the url
 function expose_prometheus() {
 	if [ "${cluster_type}" == "minikube" ]; then
-		exposed=$(kubectl get svc -n monitoring | grep "prometheus-svc")
+		exposed=$(kubectl get svc -n monitoring | grep "prometheus-test")
 		
 		# Check if the service already exposed, If not then expose the service
 		if [ -z "${exposed}" ]; then
-			kubectl expose service prometheus-k8s --type=NodePort --target-port=9090 --name=prometheus-svc -n monitoring >> ${LOG} 2>&1
+			kubectl expose service prometheus-k8s --type=NodePort --target-port=9090 --name=prometheus-test -n monitoring >> ${LOG} 2>&1
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:1453]: Replace monitoring with NAMESPACE

File: tests/scripts/common_functions.sh [hunk 7]
```diff
@@ -156,7 +156,7 @@ function deploy_autotune() {
 	# Check if the cluster_type is minikube., if so deploy prometheus
 	if [ "${cluster_type}" == "minikube" ]; then
 		echo "Installing Prometheus on minikube" >>/dev/stderr
-		setup_prometheus >> ${SETUP_LOG} 2>&1
+		setup_prometheus >> ${AUTOTUNE_SETUP_LOG} 2>&1
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:159]: I don't see prometheus cleanup being called anywhere, can you please check and invoke that at required places

File: tests/scripts/common_functions.sh [hunk 8]
```diff
@@ -174,38 +174,48 @@ function deploy_autotune() {
...
+
 # Remove the autotune setup
 # output: Remove all the autotune dependencies
 function autotune_cleanup() {
+	RESULTS_LOG=$1
+	
+	# If autotune cleanup is invoke through -t option then setup.log will inside the given result directory
+	if [ ! -z "${result_dir}" ]; then
+		AUTOTUNE_SETUP_LOG="${RESULTS_LOG}/autotune_setup.log"
+		AUTOTUNE_REPO="${AUTOTUNE_REPO}/autotune"
```
Comment (chandrams) [inline tests/scripts/common_functions.sh:205]: Isn't autotune repo already initialized on top?

Comment (shruacha1234) [reply tests/scripts/common_functions.sh:205]: when we call cleanup using -t option current directory will be in tests  so autotune_repo at the top will get initialized to 2 directory before.
```
CURRENT_DIR="$(dirname "$(realpath "$0")")"
AUTOTUNE_REPO="${CURRENT_DIR}/../.."
```

**Review by chandrams — APPROVED** [review]
LGTM

**Comment by dinogun** [comment]
@chandrams Need to fix the conflicts

**Review by dinogun — APPROVED** [review]

**Pull request merged**